modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/ConditionalConfigWorkflowOperationHandler.java

```diff
@@ -0,0 +1,108 @@
+import org.opencastproject.workflow.api.WorkflowInstance;
+import org.opencastproject.workflow.api.WorkflowOperationException;
+import org.opencastproject.workflow.api.WorkflowOperationInstance;
+import org.opencastproject.workflow.api.WorkflowOperationResult;
+import org.opencastproject.workflow.api.WorkflowOperationResult.Action;
+import org.opencastproject.workflow.conditionparser.WorkflowConditionInterpreter;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ConditionalConfigWorkflowOperationHandler extends AbstractWorkflowOperationHandler {
+  public static final String CONFIGURATION_NAME = "configuration-name";
+  public static final String CONDITION_PREFIX = "condition-";
+  public static final String VALUE_PREFIX = "value-";
+  public static final String NO_MATCH = "no-match";
+
+  private static final Logger logger = LoggerFactory.getLogger(ConditionalConfigWorkflowOperationHandler.class);
+
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+
+    Map<String, String> properties = new HashMap<>();
+
+    // Get name of wf configuration to be set
+    String wfConfigName = operation.getConfiguration(CONFIGURATION_NAME);
+    if (StringUtils.isEmpty(wfConfigName)) {
+      logger.warn("No workflow configuration was set because {} was not passed. Operation skipped.",
+              CONFIGURATION_NAME);
+      return createResult(Action.SKIP);
+    }
+
+    String value = "false";
+    // Loop through all "condition-" keys
+    for (String key : operation.getConfigurationKeys().stream().filter(x -> x.startsWith(CONDITION_PREFIX))
+            .sorted().collect(Collectors.toList())) {
+      // Evaluate condition
+      String condition = operation.getConfiguration(key);
+      if (StringUtils.isEmpty(condition)) {
+        continue;
+      }
+      boolean result = false;
+      try {
+        result = WorkflowConditionInterpreter.interpret(condition.trim());
+        logger.debug("Evaluate condition: {}, result: {}", condition, result);
+      } catch (IllegalArgumentException e) {
+        logger.warn("Invalid condition to evaluate: {}.", condition);
+        throw new WorkflowOperationException(String.format("Invalid condition: %s", condition.trim()), e);
+      }
+      // If condition true, set variable and return
+      if (result) {
+        String valuePropName = VALUE_PREFIX + key.substring(CONDITION_PREFIX.length());
+        value = operation.getConfiguration(valuePropName);
+        if (StringUtils.isEmpty(value)) {
+          logger.warn("Condition {}  evaluated to true, but no value informed to set {} variable in {}",
```

Comment:
This should be an error.
```suggestion
          logger.error("Condition {}  evaluated to true, but no value informed to set {} variable in {}",
```